### ISSUE TITLE
does cython support #define or compile directives?

### ISSUE THREAD

**Issue by lucasRogerioOliveira** [cython/cython#3249]
Hello Cythonians.

I'm trying to use cython to  use the [clone](https://linux.die.net/man/2/clone) package but there is some flags in this function that needs to define the `_GNU_SOURCE` such as: CLONE_NEWNET, CLONE_NEWNS, CLONE_NEWUTS and all other flags that needs of `CAP_SYS_ADMIN`. And I can't to use the `#define _GNU_SOURCE 1` in any kind of way that I tried. I already tried to put it into a `enum`, add it in a header file (`.h` file),  and add the option `define_macros` in the `setup.py` file but without success.

So the question is: There is some way to define it into the file? If I run this project in the root mode all works as expected, but just in the root mode.

Thank you for all kind of help and may the force be with you.

**Comment by scoder** [comment]
http://docs.cython.org/en/latest/src/userguide/external_C_code.html#including-verbatim-c-code

**Comment by lucasRogerioOliveira** [comment]
Just for curiosity, when I added this header the program still doesn't working without root mode. So, after some searches I found that I needed to use the `capabilities` command of Linux Kernel, and with this command set a` CAP_SYS_ADMIN` flag to the generated program. After that it worked. If you'll run a python script the python instance needs to have this flag, turned on.

There is a great post explaining about it, maybe it can be helpful as it was for my: https://blog.container-solutions.com/linux-capabilities-in-practice
